stl/inc/functional

```diff
@@ -1135,12 +1135,15 @@ public:
     }
 
 #if _USE_FUNCTION_INT_0_SFINAE
-    template <class _Fx, typename _Mybase::template _Enable_if_callable_t<_Fx&, function> = 0>
+    template <class _Fx, typename _Mybase::template _Enable_if_callable_t<decay_t<_Fx>&, function> = 0>
 #else // ^^^ _USE_FUNCTION_INT_0_SFINAE // !_USE_FUNCTION_INT_0_SFINAE vvv
-    template <class _Fx, class = typename _Mybase::template _Enable_if_callable_t<_Fx&, function>>
+    template <class _Fx, class = typename _Mybase::template _Enable_if_callable_t<decay_t<_Fx>&, function>>

```

Comment:
Apologies for the delayed review - I think I initially looked at this, noticed it didn't match the Standardese, and concluded that I had to write up a lengthy explanation of the metaprogramming (which I never found time for).

Upon looking at this again, I believe that it is correct - although WG21-N4901 20.14.17.3.2 [func.wrap.func.con]/9.1 says `is_same_v<remove_cvref_t<F>, function>` and this says (via `_Enable_if_callable_t`) `is_same<decay_t<decay_t<_Fx>&>, _Function>`, they have the same effect. (`decay_t` is "stronger" than `remove_cvref_t` in that decay changes arrays to pointers and functions to function pointers, but neither affects whether something is a `std::function`.) I additionally observe that it's consistent with what `operator=` is currently doing:

https://github.com/microsoft/STL/blob/18019346af33d90fab4e65a2f2a2efe741ec050b/stl/inc/functional#L1102-L1107

I'm not concerned about throughput impact since the compiler optimizes both `decay_t` and `remove_cvref_t` including our pre-C++20 helper `_Remove_cvref_t`. That said, while I think it'd be okay to proceed as-is, I think it would be slightly better to clean up `_Enable_if_callable_t`, now that LWG-2774 has added the protection against constraint recursion to the Standardese. (Originally the logic to avoid `std::function` itself was something that we added beyond the Standard.) Finally, although I am somewhat wary of modifying the old `_HAS_FUNCTION_ALLOCATOR_SUPPORT` machinery, updating that overload accordingly would make everything consistent.